Saule/Http/AllowsManuallyHandledQueryAttribute.cs

```diff
@@ -0,0 +1,54 @@
+﻿using System.Linq;
+using System.Net.Http;
+using System.Web.Http.Controllers;
+using System.Web.Http.Filters;
+using Saule.Queries;
+using Saule.Queries.Filtering;
+using Saule.Queries.Including;
+using Saule.Queries.Sorting;
+
+namespace Saule.Http
+{
+    /// <summary>
+    /// Indicates that filter, paging, include and sorting should be parsed. But they won't be automatically applied to WebApi action
+    /// and action should handle them manually
+    /// </summary>
+    public class AllowsManuallyHandledQueryAttribute : ActionFilterAttribute
```

Comment:
I wonder if this attribute is needed at all? Perhaps it's enough to detect a `QueryContext` parameter? I'm not sure if that's possible and if that would still cover the same use cases.

In any case maybe `HandlesQuery` or `ManuallyHanldesQuery` might be a nicer name?

Reply:
i renamed it to HandlesQuery. We can detect this parameter, but we still would need to initialize Sort\Filter\Include like we do in `AllowsQueryAttribute`. One way would be to move that detection to `AllowsQueryAttribute`, and then it will process them automatically or won't do anything depending on `QueryContext` parameter in action. But `AllowsQueryAttribute` might be a bit confusing as for me.

There is another point, that `HandlesQuery` attribute gives a bit more flexiblity, so developer can use `QueryContext` in action, and Saule still can do automatic IQueryable processing if needed. So it's up to developer, but yeah, additional attribute confusing it a bit. One point for future that might makes sense is to combine `AllowsQueryAttribute` and `HandlesQueryAttribute` and also add such enum to attribute constructor:

```[Flags]
public enum AutomaticProcessing
{
	None = 0,
	Sort = 1,
	Pagination = 2,
	Include = 4,
	Filter = 8,
	All = Sort | Pagination | Include | Filter
}```

So we can specify it in such way:

```[AllowsQuery(AutomaticProcissing.Filter)]
public IHttpActionResult Get()
{
	
}```

and it will initialize all query context properties from request, but will apply only filtering on action result. And all other actions won't be applied. So it would have more precise way to configure what needs to be processed and what not. And then in my case i can specify `AllowsQuery(AutomaticProcissing.None)` that will be equivalent to current `HandlesQueryAttribute` and then handle `QueryContext` in action itself.

If you like this idea, i can update the code.

Reply:
I do like it, but I also feel that the current approach is probably simpler. If we ever do a Saule 2.0 I'll consider cleaning up the attribute situation.